docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -0,0 +1,366 @@
+
+To simplify this procedure, a SHOW statement should be provided to query the effective rule for one specified object.
+
+The statement is in such a format:
+
+```sql
+SHOW PLACEMENT FOR TABLE {table_name | partition_name};
+SHOW PLACEMENT FOR INDEX index_name FROM {table_name | partition_name};
+```
+
+TiDB will automatically find the effective rule based on the rule priorities.
+
+The output of this statement contains these fields:
+
+* target: The object queried. It can be a table, partition, or index. For table and partition, it is shown in the format `TABLE table_name`; For index, it is shown in the format `INDEX index_name FROM table_name`.
+* placement: The original ALTER statement that affects the placement of this rule. Note that the object in this ALTER statement may be different with `target`. In the example above, if `target` is `p0`, the ALTER statement shown may be `ALTER TABLE t …`.
+
+### Show create table
+
+It’s useful to show rules in `SHOW CREATE TABLE` statement, because users can check the rules easily.
+
+Since data in TiDB can be imported to MySQL, the placement rules definition must be shown as a MySQL-compatible comment in such a format `/*T![placement] placement_gammar*/`, where `placement_gammar` can be recognized by TiDB. That means TiDB needs to support two approaches to define placement rules, one in `CREATE TABLE` and another in `ALTER`.
+
+This is complicated, and `ALTER` is able to satisfy most of the cases, so `SHOW CREATE TABLE` can be kept untouched so far.
+
+## Privilege management
+
+Privilege management is the same as before:
+
+* Executing `ALTER` statement requires `Alter` privilege
+* Querying `information_schema.placement_rules` only returns those placement rules on the tables that visible to the current user
+* `ADMIN SHOW DDL` statement can only be executed by those users who have `Super` privilege.
+
+## Ecosystem tools
+
+There’re many tools that are based on binlog or metadata, such as DM, Lighting, Syncer, etc. Placement rules need to be compatible with these tools.
+
+If the downstream is not TiDB, no change needs to be made. But even if it is TiDB, TiKV nodes may have a different geographical topology, which means the labels of TiKV nodes may be different. In this case, placement rules can not be enforced on them.
+
+Based on this consideration, placement rules need not to be exported to binlog or metadata. Also, placement rules are not written to binlog.
+
+However, there may be also cases where users want exactly the same placement rules as the upstream, and altering placement rules manually is very annoying. It will be considered in the future if there’s a need.
```

Comment:
the same seems to applicable for CDC?

also, does BR need to backup the placement rules? (i guess no)

Reply:
No. Addressed.